### PR TITLE
Data store

### PR DESCRIPTION
**Pull request by amdtech**
This moves the storage library into its own script, and gives access to it through `robot.storage`. I'm not sure the `migrate` function works, but initial testing shows all the regular storage commands work as expected.

Tagging @Drewzar for :eyes:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "fixed-array":            "1.0.0",
     "github":                 "0.1.16",
     "htmlparser":             "*",
-    "hubot":                  ">= 2.6.0 < 3.0.0",
+    "hubot":                  "2.11.1",
```
Comment (Drewzar) [inline package.json]: :smiley:

File: scripts/deploy-lock.coffee [hunk 2]
```diff
@@ -512,6 +457,9 @@ module.exports = (robot) ->
     history = deployers.history()
 
     messages = []
+    if _.isEmpty(history)
+      messages.push 'No deployment history!'
```
Comment (Drewzar) [inline scripts/deploy-lock.coffee]: Nice touch. :+1:

File: package.json (hunk 1 above)
Comment (amdtech) [inline package.json]: #sneakyupgrade

File: scripts/deploy-lock.coffee (hunk 2 above)
Comment (amdtech) [inline scripts/deploy-lock.coffee]: And sadly probably never seen :cry:

**Comment by Drewzar** [comment]
Looks :grapes:! What other scripts need converting other than `deployer.coffee`?

**Comment by amdtech** [comment]
Looks like it's that, and the `heroku-deploys` script has an older version of the class. I'm inclined to delete the heroku script anyway, it's not being used for anything.

**Comment by Drewzar** [comment]
lol I don't think I've ever even looked at that script :smiley_cat:

**Pull request merged**